source/common/http/http1/codec_impl.h

```diff
@@ -192,10 +192,12 @@ class ConnectionImpl : public virtual Connection, protected Logger::Loggable<Log
 
   CodecStats& stats() { return stats_; }
 
+  const bool enable_trailers_;
```

Comment:
Can you reformat all the bools in this class as a bit field? It would be nice to not waste the extra RAM for each connection.

Reply:
Just to be sure, you are talking about 
`wantsToWrite`
`supports_http_10`
`enable_trailers_`

protected:
`handling_upgrade_`
private:
`connection_header_sanitization_`
`strict_header_validation_`
`reset_stream_called_`

Which would reduce it from 7 bytes into one 4 byte bitfield.

I'll probably throw this into a separate PR.

Reply:
@mattklein123 ^^

Reply:
The number of bytes the existing fields take is dependent on ordering and alignment. It may very well be significantly larger than 5 bytes. The main issue here is we are trying to be very stringent on memory usage for things like connections. IMO this is a trivial change so I see no reason not to do it as part of this change while you wait for reviews from @alyssawilk and @PiotrSikora

Reply:
Change is being done in this PR https://github.com/envoyproxy/envoy/pull/9124 

I'll make sure to align the enable_trailers once that PR is merged in.